modules/signals/src/signal-store-assertions.ts

```diff
@@ -0,0 +1,29 @@
+import { InnerSignalStore } from './signal-store-models';
+
+declare const ngDevMode: unknown;
+
+export function assertUniqueStoreMembers(
+  store: InnerSignalStore,
+  newMemberKeys: string[]
+): void {
+  if (!ngDevMode) {
```

Comment:
Can we wrap this whole check inside the `ngDevMode` block so it gets tree-shaken during the prod build?

Reply:
Tree-shaking works with existing implementation as well. Tested on `standalone-app`.